utils/rate/gauge.go

```diff
@@ -8,17 +8,17 @@ import (
 
 // Gauge represents an exponentially-weighted moving average of given values
 type Gauge struct {
-	input metrics.Meter
-	count metrics.Meter
+	input *metrics.Meter
+	count *metrics.Meter
 	max   int64
 }
 
 // NewGauge constructs a new Gauge and launches a goroutine.
 // Be sure to call Stop() once the meter is of no use to allow for garbage collection.
 func NewGauge() *Gauge {
 	return &Gauge{
-		input: metrics.NewMeterForced(),
-		count: metrics.NewMeterForced(),
+		input: metrics.NewMeter(),
+		count: metrics.NewMeter(),
```

Comment:
The reason for NewMeterForced() is (or at least was), we need this gauge for emitter to work.
Regular NewMeter will not work if metrics collection is not enabled by using `--metrics` in os.Args.
(see https://github.com/0xsoniclabs/go-ethereum/blob/main/metrics/metrics.go#L28)
If we don't need Gauge for the emitter anymore, we can remove this Gauge struct completely.

Reply:
The metric monitoring was refactored in go-ethereum, and if I understand correctly, a single metric processing go-routine is now managing metric updates in the background. Force-enabling this for this Gauge is not trivial.

Considering that this Gauge is no longer used in the emitter on the current network (we use a fixed-rate emitter logic), I would suggest to remove the Gauge all-together and eliminate this dependency of the emitter code to metrics.

I will prepare a PR testing out how extensive the changes would be to remove the Gauge from the code base.

Reply:
The removal of the old heuristic is under evaluation in #27. Looks like this is not too difficult, but it requires some more evaluations.